File: app/src/main/java/com/bedboy/ufebri/doggie/ui/ImagesAdapter.java
package com.bedboy.ufebri.doggie.ui;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bedboy.ufebri.doggie.R;
import com.bedboy.ufebri.doggie.data.source.local.entity.DoggieEntity;
import com.bedboy.ufebri.doggie.databinding.ItemImageBinding;
import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.target.Target;

import java.util.List;

/**
 * Created by user on 5/7/18.
 */

public class ImagesAdapter extends RecyclerView.Adapter<ImagesAdapter.Holder> {
    private List<DoggieEntity> imagesGrid;

    public ImagesAdapter(List<DoggieEntity> imagesGrid) {
        this.imagesGrid = imagesGrid;
    }


    @NonNull
    @Override
    public ImagesAdapter.Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemImageBinding binding = ItemImageBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new Holder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ImagesAdapter.Holder holder, int position) {
        holder.bind(position);
    }

    @Override
    public int getItemCount() {
        return imagesGrid.size();
    }

    public class Holder extends RecyclerView.ViewHolder {

        final ItemImageBinding binding;

        Holder(ItemImageBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        void bind(int position) {

            Glide.with(itemView.getContext())
                    .load(imagesGrid.get(position).getLink())
                    .error(R.drawable.ic_launcher_background)
                    .diskCacheStrategy(DiskCacheStrategy.ALL)
                    .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
                    .placeholder(R.drawable.ic_launcher_foreground)
                    .into(binding.ivAnimal);
        }
    }
}

File: app/src/main/java/com/raylabs/doggie/ui/ImagesAdapter.kt
package com.raylabs.doggie.ui

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.request.target.Target
import com.raylabs.doggie.R
import com.raylabs.doggie.data.source.local.entity.DoggieEntity
import com.raylabs.doggie.databinding.ItemImageBinding

class ImagesAdapter(
    private val imagesGrid: List<DoggieEntity>,
    private val listener: (DoggieEntity) -> Unit
) : RecyclerView.Adapter<ImagesAdapter.Holder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val binding = ItemImageBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return Holder(binding)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        holder.bind(imagesGrid[position], listener)
    }

    override fun getItemCount(): Int = imagesGrid.size

    class Holder(private val binding: ItemImageBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bind(item: DoggieEntity, listener: (DoggieEntity) -> Unit) {
            Glide.with(itemView.context)
                .load(item.link)
                .error(R.drawable.outline_broken_image_24)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
                .placeholder(R.drawable.ic_launcher_foreground)
                .into(binding.ivAnimal)

            itemView.setOnClickListener { listener(item) }
        }
    }
}
